test/test_io.py

```diff
@@ -25,12 +25,41 @@ def test_shutdown_complete(self):
         del event_loop_group
         self.assertTrue(shutdown_event.wait(TIMEOUT))
 
+    def test_init_defaults_singleton(self):
+        event_loop_group = EventLoopGroup.get_or_create_static_default()
+        EventLoopGroup.release_static_default()
+
+    def test_init_defaults_singleton_is_singleton(self):
+        event_loop_group_one = EventLoopGroup.get_or_create_static_default()
+        event_loop_group_two = EventLoopGroup.get_or_create_static_default()
+        self.assertTrue(event_loop_group_one == event_loop_group_two)
+        EventLoopGroup.release_static_default()
+
+    def test_shutdown_complete_singleton(self):
+        event_loop_group = EventLoopGroup.get_or_create_static_default()
+        shutdown_event = event_loop_group.shutdown_event
+        del event_loop_group
+        EventLoopGroup.release_static_default()
+        self.assertTrue(shutdown_event.wait(TIMEOUT))
+
 
 class DefaultHostResolverTest(NativeResourceTest):
     def test_init(self):
         event_loop_group = EventLoopGroup()
         host_resolver = DefaultHostResolver(event_loop_group)
 
+    def test_init_singleton(self):
```

Comment:
Please add a test for MQTT where we let it use the singletons, to be sure it all works end to end in a complex case?